### PR TITLE
Voting: allow non-connected accounts to vote

### LINKED ISSUE #588: Agent: withdrawal funds
As an Aragon DAO participant or group of participants, I would like to manage assets by creating a new transfer so I can withdrawal funds in the Vault app.

Please refer to the [project PRD](https://www.notion.so/Agent-app-PRD-Product-Requirement-Document-06cfd6944f4e4c58ad9bb4fe3877d618) for more detailed information.

### LINKED ISSUE #655: Voting: display voting buttons for non-connected accounts
**Describe the bug**
The Aragon Dapp is not detecting Metamask when the Private mode is enabled.

**Mainnet or testnet?**
Mainnet

**Organization**
governance.aragonproject.eth

**To Reproduce**
1. Enable Private mode on Metamask
2. Try voting.

**Expected behavior**
1. A Metamask window should pop-up asking whether I want Metamask to interact with the current page/Dapp.
2. Once approved, the Dapp should be able to communicate with Metamask.

**Additional context**
The Aragon Dapp is among the last Dapps to have no support for this very important privacy feature. (🤦‍♂️)

### PR DESCRIPTION
**Pull request by bpierre**
Fixes #655

Until we get the connect button accessible from anywhere.

### CODE REVIEW

File: apps/voting/app/src/components/VotePanelContent.js [hunk 1]
```diff
@@ -312,16 +319,19 @@ class VotePanelContent extends React.Component {
...
-                      You will cast your vote with{' '}
-                      {userBalance === null
-                        ? '… tokens'
-                        : pluralize(userBalance, '$ token', '$ tokens')}
-                      , since it was your balance at the beginning of the vote{' '}
-                      {this.renderBlockLink(snapshotBlock)}.
-                    </p>
-                  </Info.Action>
+                  {/* This check is until we get the connect button in the MenuPanel */
+                  userBalance !== null && (
```
Comment (sohkai) [inline apps/voting/app/src/components/VotePanelContent.js]: Maybe we could detect that there's no account and instead say "You will need to connect your account to vote"?

File: apps/voting/app/src/components/VotePanelContent.js [hunk 2]
```diff
@@ -94,17 +94,24 @@ class VotePanelContent extends React.Component {
...
-      app
-        .call('canVote', vote.voteId, user)
-        .first()
-        .subscribe(canVote => {
-          this.setState({ loadingCanVote: false, userCanVote: canVote })
-        })
+    // Note: the next section is commented until we get the connect button in
+    // the MenuPanel. Until then, the user need to access the signing panel
+    // in order to connect, so we consider that the user can vote.
+    if (vote) {
```
Comment (sohkai) [inline apps/voting/app/src/components/VotePanelContent.js]: I had a few thoughts on this, especially in the context of responsive screens: https://github.com/aragon/aragon/issues/315#issuecomment-461778318.

I think we can allow the user to vote if their account isn't connected, but still check that they can actually vote if an account is available.

File: apps/voting/app/src/components/VotePanelContent.js (hunk 1 above)
Comment (bpierre) [reply apps/voting/app/src/components/VotePanelContent.js]: Yes good idea! Also, what would you think about a way for apps to request the account connection?

Comment (sohkai) [reply apps/voting/app/src/components/VotePanelContent.js]: We could certainly bake that into aragon.js 👍 : https://github.com/aragon/aragon.js/issues/237

File: apps/voting/app/src/components/VotePanelContent.js [hunk 3]
```diff
@@ -104,6 +108,10 @@ class VotePanelContent extends React.Component {
         .subscribe(canVote => {
           this.setState({ loadingCanVote: false, userCanVote: canVote })
         })
+    } else {
+      // Note: if the account is not present, we assume the account it’s not
```
Comment (sohkai) [inline apps/voting/app/src/components/VotePanelContent.js]: Nitpick: ~~it's~~ **is**

**Pull request merged**